src/AsyncTCP.cpp

```diff
@@ -546,7 +552,25 @@ AsyncClient::AsyncClient(tcp_pcb* pcb)
 , next(NULL)
 {
     _pcb = pcb;
+    _closed_slot = -1;
     if(_pcb){
+        _closed_slot = 0;
+        if (_closed_index == 0) {
+            _closed_index = 1;
+            for (int i = 0; i < _number_of_closed_slots; ++ i) {
+                _closed_slots[i] = 1;
+            }
+        } else {
+            int closed_slot_min_index = _closed_slots[0];
+            for (int i = 0; i < _number_of_closed_slots; ++ i) {
+                if (_closed_slots[i] <= closed_slot_min_index && _closed_slots[i] != 0) {
+                    closed_slot_min_index = _closed_slots[i];
+                    _closed_slot = i;
+                }
+            }
+        }
+        _closed_slots[_closed_slot] = 0;
+
```

Comment:
this whole block and anywhere that you access `_closed_slots` should be protected by semaphore. else creating clients from different places might cause concurency issues :)